src/component/NodeEditor/NodeEditor.tsx

```diff
@@ -385,6 +386,12 @@ export class NodeEditor extends React.PureComponent<NodeEditorProps> {
     constructor(props: NodeEditorProps) {
         super(props);
 
+        const showResultName = props.resultName.length > 0;
+
+        this.state = {
+            showResultName
```

Comment:
thought you were going to keep this in the redux store?

Reply:
oh sorry i thought you had suggested keeping that in `NodeEditor`, will move it back

Reply:
I think since we are so close, just sticking to the policy of all state in redux for now (but scoped) makes sense. We can revisit this later, but I think this just reduces the cognitive load for someone coming into the project.